src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Taxonomy.java

```diff
@@ -31,14 +34,20 @@
 @Entity
 public class Taxonomy extends Auditable<Long> {
 
-    @ApiModelProperty(position = 1, example = "1")
+    @ApiModelProperty(position = 1, value = "Taxonomy auto generated id", required = true, readOnly = true)
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @ApiModelProperty(position = 2, example = "1")
     @NotNull
     @JsonProperty
-    @Id
     @Min(1)
-    private long id;
+    @Column(unique = true)
+    private long taxonomyIdentifier;
```

Comment:
I would prefer `taxonId` as it is shorter but still clear. It is just my opinion you don't have to change it.

Reply:
Given that we are using 'id' everywhere else, I would go for 'taxonomyId'. I would prefer not to shorten other words unless absolutely necessary.